### PR TITLE
Minor refactor to prefer C++ Standard Algorithms

### PR DESCRIPTION
**Pull request by tcoyvwac**
For translation unit: `StorageSystemZooKeeper`:

* Refactored some `ranged-for` loops to their standard library's equivalent.
  * _Long-term maintainability and testing of the codebase improves as the standard library abstraction is supported (by the C++ standards committee) for many future years ahead._

---
(EDIT: So, I have not so much personal preference in styles btw,) this PR only follows the suggestions for projects from:
* Sonarsource static analysis website: ["STL algorithms and range-based for loops should be preferred to traditional for loops"](https://rules.sonarsource.com/cpp/RSPEC-5566/) .
* On the recommended C++ Core Guidelines website:  [SL.1: Use libraries wherever possible](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rsl-lib), and especially [SL.2: Prefer the standard library to other libraries](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rsl-sl).
  _This PR's text is a more neutral version of SL.2's text._ :sweat_smile: 

---
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Storages/System/StorageSystemZooKeeper.cpp [hunk 1]
```diff
@@ -285,10 +288,12 @@ void StorageSystemZooKeeper::fillData(MutableColumns & res_columns, ContextPtr c
             }), nodes.end());
         }
 
-        std::vector<std::future<Coordination::GetResponse>> futures;
-        futures.reserve(nodes.size());
-        for (const String & node : nodes)
-            futures.push_back(zookeeper->asyncTryGet(path_part + '/' + node));
+        std::vector<std::future<Coordination::GetResponse>> futures(nodes.size());
```
Comment (antonio2368) [inline src/Storages/System/StorageSystemZooKeeper.cpp:291]: `reserve` and construtor accepting count are not the same thing.
Your implementation will defaulkt construct `nodes.size()` object while `reserve` will just reserve enough space for that amount of objects.
`reserve` should be returned.

**Comment by tcoyvwac** [comment]
Thanks @antonio2368 for merging! Grateful! :tada:

**Comment by alexey-milovidov** [comment]
Sorry, but the code clearly became worse.

C++ standard algorithms were popular about 10 years ago and only inside a limited circle of FP-fans.
Now it's a different time... and for loop almost always better.

**Comment by tcoyvwac** [comment]
I see, sorry, maybe I am missing something but was there an error with the code? Trying to understand a bit more why the code was reverted. 
* If there was an error, sorry the blame is with me! I am trying to understand what would it be. It must have slipped by this code review and will improve for next time. 
* If it because of stylistic reasons then that's no problem for the revert, and I can relax a bit more knowing things are ok! :smile:

**Comment by alexey-milovidov** [comment]
No problem, no need to worry!

The modification was correct.
But it did not improve the code. It increaded the number of lines (which is not always bad btw) but also made the code more difficult to read. We asked a few developers from my team to check what variant is more easy to understand and the old variant have won.

**Comment by tcoyvwac** [comment]
Thanks for the writing @alexey-milovidov and greetings to the team. Your constructive inside-view behind the decision really helps understand the reason for the revert! 
:+1: For sure, if the style / coding variant isn't preferable for the project or team (because the code-review missed something), I really agree and relate with the decision to revert. 

It really helps a lot contributors during code-review, however now understand the team's preference so I will adjust for next time & future PRs. :smile: Thanks, ye!

**Pull request merged**